src/main/java/io/fabric8/maven/docker/service/RunService.java

```diff
@@ -464,11 +467,23 @@ private void startContainer(ImageConfiguration imageConfig, String id, GavLabel
     }
 
     private void updateMappedPortsAndAddresses(String containerId, PortMapping mappedPorts) throws DockerAccessException {
-        Container container = queryService.getMandatoryContainer(containerId);
-        if (container.isRunning()) {
-            mappedPorts.updateProperties(container.getPortBindings());
-        } else {
-            log.warn("Container %s is not running anymore, can not extract dynamic ports",containerId);
+        for (int attempt = 0; attempt < UPDATE_MAPPED_PORTS_RETRIES && !Thread.currentThread().isInterrupted(); attempt++) {
+            try {
+                Container container = queryService.getMandatoryContainer(containerId);
+                if (container.isRunning()) {
+                    mappedPorts.updateProperties(container.getPortBindings());
+                } else {
+                    log.warn("Container %s is not running anymore, can not extract dynamic ports", containerId);
+                }
+                return;
+            } catch (PortBindingException e) {
+                try {
+                    log.warn("Failed to get port bindings for container %s (attempt %d of %d)", containerId, attempt, UPDATE_MAPPED_PORTS_RETRIES);
```

Comment:
i would change this to debug, but add at the end a error log if this fails.

Reply:
yeah, I think thats sensible. Have also added a debug log message when it succeeds too.